### DISCUSSION TITLE
Receiving the dictionary in the ViewModel when sent with Shell.Current.GoToAsync

### DISCUSSION THREAD

**Discussion by Orgbrat** [dotnet/maui#4968] [Q&A] [answered]
So passing shell parameters of complex objects was added with [PR #2004 ](https://github.com/dotnet/maui/pull/2004) in August of 2021 using a syntax similar to below.  But my question is how do you get the Dictionary in the called ViewModel. I am usiing QueryProperty attributes on my ViewModels already.  is the Dictionary just another QueryProperty on the ViewModel?

When you navigate with Shell you can pass IDictionary<string,object> in the method, just like:

await Shell.Current.GoToAsync("//myAwesomeUri", new Dictionary { {"My key", someObj} });

Orgbrat

**Comment by pictos** [comment, marked as answer]
@Orgbrat maybe this approach will be better for your case
https://docs.microsoft.com/en-us/xamarin/xamarin-forms/app-fundamentals/shell/navigation#process-navigation-data-using-a-single-method

**Reply by Orgbrat** [reply]
Thanks @pictos , your recommended approach was just what I needed. At first I was worried because the docs  showed a dictionary of <string, string> but when I started coding this approach the dictionary passed thru in MAUI was the dictionary of <string, object>. And this works just perfect for what I needed.  Again thanks for the help and it was much appreciated.

**Reply by pictos** [reply]
@Orgbrat you're welcome. Yeah the docs should be using the Xamarin.Forms API, which doesn't allow <string,object>
